src/components/paginator-render/paginator-render.tsx

```diff
@@ -0,0 +1,668 @@
+    }
+  };
+
+  #handleGoToOnBlur = (event: FocusEvent) => {
+    this.#handleGotoPageChange(event);
+  };
+
+  #handleGotoPageChange = (event: Event) => {
+    const input = event.target as HTMLInputElement;
+
+    const page = Number(input.value);
+
+    if (this.totalPages < 0) {
+      this.pageChange.emit(page);
+    }
+
+    // Validate if page exceeds total pages range
+    if (page >= 1 && page <= this.totalPages) {
+      this.actualPage = Math.max(Math.min(page, this.totalPages), 1);
+      this.pageChange.emit(this.actualPage);
+    }
+  };
+
+  #handleSelectedPageChange = (event: PointerEvent) => {
+    event.stopPropagation();
+    const elementRef = event.composedPath()[0] as
+      | HTMLAnchorElement
+      | HTMLButtonElement;
+
+    // Check the click event is performed in a button and anchor element
+    if (
+      elementRef.tagName.toLowerCase() !== "button" &&
+      elementRef.tagName.toLowerCase() !== "a"
+    ) {
+      return;
+    }
+    event.preventDefault();
+
+    let newPageSelected;
+
+    if (elementRef.tagName.toLowerCase() === "a") {
+      newPageSelected = this.hasHyperlink
+        ? (elementRef as HTMLAnchorElement)?.href
+        : elementRef?.id;
+    } else {
+      newPageSelected = (elementRef as HTMLButtonElement)?.value;
+    }
+
+    // Don't fire the selectedItemChange event if the button is already selected
+    if (this.selectedPage === newPageSelected) {
+      return;
+    }
+    this.selectedPage = newPageSelected;
+
+    const eventInfo = this.pageChange.emit(this.actualPage);
+
+    if (!eventInfo.defaultPrevented) {
+      this.selectedPage = newPageSelected;
+    }
+  };
+
+  #renderItemsPerPage = () => {
+    return (
+      <form
+        class="form"
+        part={tokenMap({ [PAGINATOR_PARTS_DICTIONARY.ITEMS_PER_PAGE]: true })}
+      >
+        <label htmlFor="items-per-page">
+          {this.translations.text.itemsPerPage}
+        </label>
+        <ch-combo-box-render
+          class="combo-box"
+          id="items-per-page"
+          name="items-per-page"
+          model={this.itemsPerPageOptions}
+          onChange={this.#handleItemsPerPageChange}
+          hostParts={tokenMap({
+            [PAGINATOR_PARTS_DICTIONARY.ITEMS_PER_PAGE__COMBO_BOX]: true
+          })}
+          value={this.itemsPerPage.toString()}
+          exportparts={COMBO_BOX_EXPORT_PARTS}
+        />
+      </form>
+    );
+  };
+
+  #renderItemsPerPageInfo = () => {
+    return (
+      <div part={tokenMap({ [PAGINATOR_PARTS_DICTIONARY.INFO]: true })}>
+        <span
+          part={tokenMap({ [PAGINATOR_PARTS_DICTIONARY.INFO__TEXT]: true })}
+        >
+          {this.translations.text.showingItems
+            .replace(AMOUNT_ITEMS_IN_PAGE, this.itemsPerPage.toString())
+            .replace(
+              TOTAL_ITEMS,
+              this.totalItems.toString() || this.translations.text.unknownItems
+            )}
+        </span>
+      </div>
+    );
+  };
```

Comment:
We can use implicit return save up some bytes in the output JS.
```suggestion
  #renderItemsPerPageInfo = () => (
    <div part={PAGINATOR_PARTS_DICTIONARY.INFO}>
      <span part={PAGINATOR_PARTS_DICTIONARY.INFO__TEXT}>
        {this.translations.text.showingItems
          .replace(AMOUNT_ITEMS_IN_PAGE, this.itemsPerPage.toString())
          .replace(
            TOTAL_ITEMS,
            this.totalItems.toString() || this.translations.text.unknownItems
          )}
      </span>
    </div>
  );
```